src/shared/dbgutil/elfreader.h

```diff
@@ -67,7 +67,11 @@ class ElfReader
     bool EnumerateLinkMapEntries(ElfW(Dyn)* dynamicAddr);
 #endif
     bool EnumerateProgramHeaders(ElfW(Phdr)* phdrAddr, int phnum, uint64_t baseAddress, uint64_t* ploadbias, ElfW(Dyn)** pdynamicAddr);
+#ifdef __FreeBSD__
+    virtual void VisitModule(caddr_t baseAddress, std::string& moduleName) { };
```

Comment:
I'm curious what size the 'caddr_t` type resolves to. Addresses in createdump are always a 64bit value even on the 32 bit architectures.

Reply:
This was part two of a fix put in https://github.com/dotnet/diagnostics/pull/2419/files#diff-a22518b4eef840d3eed60b5a41049355c742d40b550d17808931323a78628a4b
with out either fix (reverting the fix in the elfreader.cpp; not using elfreader.h fix from this pr):
```
/root/diagnostics/src/shared/dbgutil/elfreader.cpp:624:33: error: cannot initialize a parameter of type 'void *' with an rvalue of type 'const char *'
                if (!ReadMemory(map.l_name + i, &ch, sizeof(ch))) {
                                ^~~~~~~~~~~~~~
/root/diagnostics/src/shared/dbgutil/elfreader.h:76:35: note: passing argument to parameter 'address' here
    virtual bool ReadMemory(void* address, void* buffer, size_t size) = 0;
                                  ^
/root/diagnostics/src/shared/dbgutil/elfreader.cpp:637:21: error: cannot initialize a parameter of type 'uint64_t' (aka 'unsigned long') with an lvalue of type 'caddr_t' (aka 'char *')
        VisitModule(map.l_addr, moduleName);
                    ^~~~~~~~~~
/root/diagnostics/src/shared/dbgutil/elfreader.h:73:39: note: passing argument to parameter 'baseAddress' here
    virtual void VisitModule(uint64_t baseAddress, std::string& moduleName) { };
```
with just the .cpp fix:

```
/root/diagnostics/src/shared/dbgutil/elfreader.cpp:638:21: error: cannot initialize a parameter of type 'uint64_t' (aka 'unsigned long') with an lvalue of type 'caddr_t' (aka 'char *')
        VisitModule(map.l_addr, moduleName);
                    ^~~~~~~~~~
/root/diagnostics/src/shared/dbgutil/elfreader.h:73:39: note: passing argument to parameter 'baseAddress' here
    virtual void VisitModule(uint64_t baseAddress, std::string& moduleName) { };
```
`caddr_t` predates sane coding logic. most things use `char  *` or `void *` now...but not `ptrace(2)` on *BSD...or OSX